scipy/signal/wavelets.py

```diff
@@ -308,14 +308,96 @@ def ricker(points, a):
     return total
 
 
-def cwt(data, wavelet, widths):
+def morlet2(M, s, w=5):
+    """
+    Complex Morlet wavelet, designed to work with `cwt`.
+
+    Returns the complete version of morlet wavelet, normalised
+    according to `s`:
+
+        ``exp(1j*w*x/s) * exp(-0.5*(x/s)**2) * pi**(-0.25) * sqrt(1/s)``
+
+    .. versionadded:: 0.18.2
+
+    Parameters
+    ----------
+    M : int
+        Length of the wavelet.
+    s : float
+        Width parameter of the wavelet.
+    w : float, optional
+        Omega0. Default is 5
+
+    Returns
+    -------
+    morlet : (M,) ndarray
+
+    See Also
+    --------
+    morlet : Implementation of Morlet wavelet, incompatible with `cwt`
+
+    Notes
+    -----
+    This function was designed to work with `cwt`. Because `morlet2`
+    returns an array of complex numbers, the `dtype` argument of `cwt`
+    should be set to `numpy.complex` for best results.
+
+    Note the difference in implementation with `morlet`.
+
+    The funamental frequency of this wavelet in Hz is given by:
+
+        ``f = w*fs / (2*s*np.pi)``
+
+    where `fs` is the sampling rate and `s` is the wavelet width parameter.
+    Similarly we can get the wavelet width parameter at `f`:
```

Comment:
same, double backticks around ``f``. the logic is: single backticks or for referencing functions or parameters, double backticks are for formatting as code snippets.